servers/visual/shader_language.cpp

```diff
@@ -614,7 +614,7 @@ ShaderLanguage::Token ShaderLanguage::_get_token() {
 
 					char_idx += str.length();
 					Token tk;
-					if (period_found || minus_exponent_found || float_suffix_found)
+					if (period_found || exponent_found || float_suffix_found)
```

Comment:
```
servers/visual/shader_language.cpp: In member function 'ShaderLanguage::Token ShaderLanguage::_get_token()':
servers/visual/shader_language.cpp:528:11: error: variable 'minus_exponent_found' set but not used [-Werror=unused-but-set-variable]
      bool minus_exponent_found = false;
           ^
```
:)

Reply:
@akien-mga I think we need to find out who introduced the minus_exponent_found logic, obviously someone on purpose made positive exponents be treated as integers but didn't test it through. My best guess is that this is a change in one of the other parsers where it makes sense and simply duplicated for the GLSL shader parser.

I'm happy to remove the minus_exponent_found logic but I left it there more or less in case someone came back with a reason why it was added in the first place.

Reply:
It was added by @reduz in 22d83bc9f6.

Reply:
Well hopefully @reduz will shed some light on it then before this gets fixed the wrong way around.

Reply:
From https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)

> Literals
> 
> A literal is a value written into the shader. Literals are always basic types, and they can be used to initialize the value of a basic type. They can also be used wherever a constant basic value is needed.
> 
> Literals are typed. There are two literals of bool type: true and false.
> 
> Any integer value is by default of type int (a signed integer). To create an integer literal of unsigned type, use the suffix u or U.
> 
> Integer literals can be defined in base 8 or base 16 using the standard C/C++ conventions (prefix with 0 for base 8 or 0x for base 16).
> 
> A numeric literal that uses a decimal is by default of type float. To create a float literal from an integer value, use the suffix f or F as in C/C++.
> 
> **Floats can also be defined using exponential notation, using the e or E to separate the base from the exponent. The exponent is always base-10.**
> 
> In GLSL 4.00 and above, double-precision floats are available. By default, all floating-point literals are of type float. To create a double-precision float, use the lf or LF suffixes. This will force it to the double type.

Especially that 2nd to last paragraph specifically mentioning floats can use exponential notation. So my guess is that checking for the negative value isn't needed. If we've detected its exponential than it's float.